controllers/deactivation/deactivation_controller.go

```diff
@@ -188,14 +227,41 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	if !found || deactivatingCondition.Status != corev1.ConditionTrue ||
 		deactivatingCondition.Reason != toolchainv1alpha1.UserSignupDeactivatingNotificationCRCreatedReason {
 		// If the UserSignup has been marked as deactivating, however the deactivating notification hasn't been
-		// created yet, then wait - the notification should be created shortly by the UserSignup controller
-		// once the "deactivating" state has been set which should cause a new reconciliation to be triggered here
+		// created yet, then set the deactivation timestamp to a temporary value, calculated as being the current
+		// timestamp plus the number of pre-deactivation days configured in the settings, BUT only if it is currently nil
+		// OR the calculated value is more than 12 hours later than the current value
+		deactivationDueTime := time.Now().Add(time.Duration(deactivatingNotificationDays) * 12 * time.Hour)
```

Comment:
shouldn't this be calculated since the notification was created instead? Also, why `* 12 * time.Hour`? Days has 24 hours if I'm not mistaken :thinking:

Reply:
ah, this is for the case when the notification is not created. but why such a value?